dask_expr/_rolling.py

```diff
@@ -0,0 +1,218 @@
+import functools
+from collections import namedtuple
+from numbers import Integral
+
+from dask.dataframe.dispatch import meta_nonempty
+
+from dask_expr._collection import new_collection
+from dask_expr._expr import Blockwise, Expr, Projection, make_meta
+
+BlockwiseDep = namedtuple(typename="BlockwiseDep", field_names=["iterable"])
+
+
+class RollingReduction(Expr):
+    _parameters = [
+        "frame",
+        "window",
+        "kwargs",
+        "how_args",
+        "how_kwargs",
+    ]
+    _defaults = {
+        "kwargs": None,
+        "how_args": (),
+        "how_kwargs": None,
+    }
+    how = None
+
+    @functools.cached_property
+    def npartitions(self):
+        return self.frame.npartitions
+
+    def _divisions(self):
+        return self.frame.divisions
+
+    @functools.cached_property
+    def _meta(self):
+        rolling = meta_nonempty(self.frame._meta).rolling(self.window, **self.kwargs)
+        meta = getattr(rolling, self.how)(*self.how_args, **self.how_kwargs or {})
+        return make_meta(meta)
+
+    @functools.cached_property
+    def kwargs(self):
+        return {} if self.operand("kwargs") is None else self.operand("kwargs")
+
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            return type(self)(self.frame[parent.operand("columns")], *self.operands[1:])
+
+    @property
+    def _has_single_partition(self):
+        return (
+            self.kwargs.get("axis") in (1, "columns")
+            or (isinstance(self.window, Integral) and self.window <= 1)
+            or self.frame.npartitions == 1
+        )
+
+    def _lower(self):
+        return RollingAggregation(
+            self.frame,
+            self._has_single_partition,
+            self.window,
+            self.kwargs,
+            self.how,
+            list(self.how_args),
+            self.how_kwargs,
+        )
+
+
+class RollingAggregation(Blockwise):
+    _parameters = [
+        "frame",
+        "_has_single_partition",
+        "window",
+        "kwargs",
+        "how",
+        "how_args",
+        "how_kwargs",
+    ]
+
+    @staticmethod
+    def operation(
+        frame, _has_single_partition, window, kwargs, how, how_args, how_kwargs
+    ):
+        if _has_single_partition:
+            rolling = frame.rolling(window, **kwargs)
+            return getattr(rolling, how)(*how_args, **(how_kwargs or {}))
+        else:
+            raise NotImplementedError("Not implemented for more than 1 partitions")
```

Comment:
We should put the npartitions> 1 case into a different class, I'd rather not mix those up, the check needs to go into the lower method above

Reply:
I moved it into RollingReduction._lower as suggested. But wasn't sure about the moving into different class. Just inherit from RollingReduction and call it RollingReductionMultiPartition(?). 
https://github.com/dask-contrib/dask-expr/pull/405/commits/5b68fbbbb4c95e4da68fd0f93a8b77aef8c8525e

Reply:
Or sorry, confused, meant dispatch to `RollingAggregationMultiPartition`?

Reply:
Yeah kind of, but we will most likely only dispatch to MapOverlap instead (see my other pr).

Anyway, this decision should be made in lower generally speaking